ui/src/components/shared/dialogs/JSONViewerDialog.vue

```diff
@@ -1,19 +1,25 @@
 <template>
   <Dialog v-model:visible="visible" modal maximizable :header="'JSON Viewer'" :style="{ width: '75vw' }">
-    <VueJsonPretty class="json" :path="'res'" :data="data" />
+    <VueJsonPretty v-if="!editMode" class="json" :path="'res'" :data="data" />
+    <JSONEditor v-else-if="editMode" v-model:data="editData" />
     <template #footer>
-      <Button label="OK" @click="visible = false" text />
+      <Button v-if="!editMode" label="Edit" @click="editMode = true" text severity="help" />
```

Comment:
is it possible to have a v-if on a template? If so 2 separate templates would be clearer.